smt/surrogate_models/krg_based.py

```diff
@@ -1253,36 +1327,65 @@ def _check_param(self):
         """
         d = self.options["n_comp"] if "n_comp" in self.options else self.nx
 
-        if len(self.options["theta0"]) != d:
-            if len(self.options["theta0"]) == 1:
-                if self.options["categorical_kernel"] in [
-                    HOMO_GAUSSIAN,
-                    FULL_GAUSSIAN,
-                ]:
-                    n_param = compute_n_param(
-                        self.options["xtypes"], self.options["categorical_kernel"]
-                    )
-                    self.options["theta0"] *= np.ones(n_param)
-                else:
-                    self.options["theta0"] *= np.ones(d)
+        if self.name in ["KPLS"]:
+            if self.options["corr"] not in ["squar_exp", "abs_exp"]:
+                raise ValueError(
+                    "KPLS only works with a squared exponential or an absolute exponential kernel"
+                )
 
-            else:
-                if not (
-                    self.options["categorical_kernel"] in [HOMO_GAUSSIAN, FULL_GAUSSIAN]
-                ):
+        if self.options["categorical_kernel"] is not None:
+            if self.options["categorical_kernel"] not in [
```

Comment:
Yes, in case of an "enumeration" type for an option, simply listing valid values is enough, no need to check the type of values which is less restrictive and redundant anyway.